app-text/texlive/texlive-2021.recipe

```diff
@@ -0,0 +1,958 @@
+# otherwise the current working directory is used.
+#
+###		SYSTEXMF = $TEXMFSYSVAR;$TEXMFLOCAL;$TEXMFDIST
+#
+# First writable tree here is used by Lua(La)TeX for the font cache.
+# LuaLaTeX uses the value here, while ConTeXt uses the same variable but
+# from texmfcnf.lua; therefore the two values should be kept in sync.
+# (As should everything else in texmf.cnf <-> texmfcnf.lua.)
+#
+###		TEXMFCACHE = $TEXMFSYSVAR;$TEXMFVAR
+#
+# Where generated fonts may be written.  This tree is used when the sources
+# were found in a system tree and either that tree wasn't writable, or the
+# varfonts feature was enabled in MT_FEATURES in mktex.cnf.
+#
+###		VARTEXFONTS = $TEXMFVAR/fonts
+#
+# On some systems, there will be a system tree which contains all the font
+# files that may be created as well as the formats.  For example
+#   TEXMFVAR = /var/lib/texmf
+# is used in many distros.  In this case, set VARTEXFONTS like this
+#VARTEXFONTS = $TEXMFVAR/fonts
+# and do not mention it in TEXMFDBS (but _do_ mention TEXMFVAR).
+#
+# Remove $VARTEXFONTS from TEXMFDBS if the VARTEXFONTS directory is below
+# one of the TEXMF directories (avoids overlapping ls-R files).
+fixKSearchPaths()
+{
+	TEXMFCNF=$sourceDir/texk/kpathsea/texmf.cnf
+	sed -i -e "s@TEXMFROOT =.*@TEXMFROOT = $dataDir/texlive@" $TEXMFCNF
+	sed -i -e "s@TEXMFDIST =.*@TEXMFDIST = $dataDir/texlive/texmf-dist@" $TEXMFCNF
+	sed -i -e "s@TEXMFMAIN =.*@TEXMFMAIN = $dataDir/texlive/texmf-dist@" $TEXMFCNF
+	sed -i -e "s@TEXMFLOCAL =.*@TEXMFLOCAL = /boot/system/non-packaged/data/texmf-dist@" $TEXMFCNF
+	sed -i -e "s@TEXMFSYSVAR =.*@TEXMFSYSVAR = $sharedStateDir/texlive@" $TEXMFCNF
+	sed -i -e "s@TEXMFSYSCONFIG =.*@TEXMFSYSCONFIG = $sysconfDir/texlive@" $TEXMFCNF
+	sed -i -e "s@TEXMFHOME =.*@TEXMFHOME = /boot/home/config/settings/texmf-dist@" $TEXMFCNF
+	sed -i -e "s@TEXMFVAR =.*@TEXMFVAR = /boot/home/config/var/texmf@" $TEXMFCNF
+	sed -i -e "s@TEXMFCONFIG =.*@TEXMFCONFIG = /boot/home/config/settings/texmf/config@" $TEXMFCNF
+	sed -i -e "s@VARTEXFONTS =.*@VARTEXFONTS = /boot/home/config/settings/texmf/fonts@" $TEXMFCNF
+	# rename current TEXMFCNF; easier than writing a more complex sed script!
+	sed -i -e "s@^TEXMFCNF@IGNOREDTEXMFCNF@" $TEXMFCNF
+	# redefine TEXMFCNF: {$SELFAUTOLOC,$SELFAUTODIR,$SELFAUTOPARENT}{,{/share,}/texmf{-local,}/web2c}
+	sed -i -e "s@%TEXMFCNF.*@TEXMFCNF={$dataDir/texlive/texmf-dist/web2c,/boot/system/non-packaged/data/texmf-dist/web2c,/boot/home/config/settings/texmf-dist/web2c}@" $TEXMFCNF
+}
+
+PATCH()
+{
+	# disable the install-bin-links target
+	sed -i -e 's/!WIN32/WIN32/g' am/bin_links.am
+
+	./reautoconf
+
+	# may need to run this super early to ensure everything gets the right value...
+	fixKSearchPaths
+}
+
+BUILD()
+{
+	mkdir -p ../build && cd ../build
+
+	if [ $targetArchitecture = x86_gcc2 ] || [ $targetArchitecture = x86 ]; then
+		HOST=i586-pc-haiku
+	elif [ $targetArchitecture = x86_64 ]; then
+		HOST=x86_64-unknown-haiku
```

Comment:
this could be HOST=$targetArchitecture-unknown-haiku and work for all non-x86-32bit architectures (when we start building packages for them)